swarm_dial.go

```diff
@@ -529,3 +593,48 @@ func (s *Swarm) dialAddr(ctx context.Context, p peer.ID, addr ma.Multiaddr) (tra
 	// success! we got one!
 	return connC, nil
 }
+
+// TODO We should have a `IsFdConsuming() bool` method on the `Transport` interface in go-libp2p-core/transport.
+// This function checks if any of the transport protocols in the address requires a file descriptor.
+func (s *Swarm) IsFdConsumingAddr(addr ma.Multiaddr) bool {
```

Comment:
I wouldn't export this.

Reply:
@Stebalien How do I unit-test it without exporting it ? 

1. All the testing helpers are in a `/testing` package which depends on go-libp2p-swarm.
2. All our tests are in `swarm_test` package.

Do you want me to write a unit-test in the swarm package even if that means copying some of the stuff in the `testing` package ?

Reply:
I thought I commented on this....

This is fine as-is (I didn't realize why we were exporting it).